dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/models/SubSnpCoreFieldsForwardStrandMappingTest.java

```diff
@@ -290,10 +302,43 @@ public void hgvsCNullHgvsTReverseStrandMustChange() throws Exception {
                                                                   91223961L,
                                                                   91223961L,
                                                                   "T", "T", "TAGA", "T/TAGA",
+                                                                  1,
                                                                   null, null, null, 1,
                                                                   "NT_455866.1:g.1766473insAGA", 1766472L, 1766472L, -1);
 
         assertEquals("A", subSnpCoreFields1.getReferenceInForwardStrand());
         assertEquals("TCTA", subSnpCoreFields1.getAlternateInForwardStrand());
     }
+
+    @Test
+    public void allelesReverseStrandMustChange() throws Exception {
+        assertEquals("T/C", buildSubSnpCoreFieldsWithOrientations("G/A", -1, 1, 1).getAllelesInForwardStrand());
+        assertEquals("T/C", buildSubSnpCoreFieldsWithOrientations("G/A", 1, -1, 1).getAllelesInForwardStrand());
+        assertEquals("T/C", buildSubSnpCoreFieldsWithOrientations("G/A", 1, 1, -1).getAllelesInForwardStrand());
+        assertEquals("T/C", buildSubSnpCoreFieldsWithOrientations("G/A", -1, -1, -1).getAllelesInForwardStrand());
+    }
+
+    @Test
+    public void allelesReverseStrandMustNotChange() throws Exception {
+        assertEquals("T/C", buildSubSnpCoreFieldsWithOrientations("T/C", 1, 1, 1).getAllelesInForwardStrand());
+        assertEquals("T/C", buildSubSnpCoreFieldsWithOrientations("T/C", -1, -1, 1).getAllelesInForwardStrand());
+        assertEquals("T/C", buildSubSnpCoreFieldsWithOrientations("T/C", 1, -1, -1).getAllelesInForwardStrand());
+        assertEquals("T/C", buildSubSnpCoreFieldsWithOrientations("T/C", -1, 1, -1).getAllelesInForwardStrand());
+    }
+
+    private SubSnpCoreFields buildSubSnpCoreFieldsWithOrientations(String alleles, int subsnpOrientation,
+                                                                   int snpOrientation,
+                                                                   int contigOrientation) {
+        return new SubSnpCoreFields(26954817, 13677177L, snpOrientation,
+                                    "NT_455866.1", 1766472L, 1766472L, contigOrientation,
+                                    "4", 91223961L, 91223961L,
+                                    "T", "T", "C", alleles, subsnpOrientation,
+                                    "NC_006091.4:g.91223961T>C", 91223961L, 91223961L, 1,
+                                    "NT_455866.1:g.1766472T>C", 1766472L, 1766472L, 1);
```

Comment:
maybe leaving those fields from an actual variant is misleading. should we put null values instead? like `SubSnpCoreFields(0, 0, snpOrientation, null, 0, 0 ...` to highlight the only fields that matter?

Reply:
I'm fine with setting irrelevant fields to null.